Realm/RLMProperty.mm

```diff
@@ -343,8 +381,11 @@ - (instancetype)initWithName:(NSString *)name
 
     _name = name;
     _indexed = indexed;
+    _objectClassName = linkingObjectsInfo[@"class"];
```

Comment:
We can't usefully validate anything at this point. Some validation is performed within `-setTypeFromRawType`, and the majority will be performed at the object store layer within `Schema::validate`.